internal/manifest/manifest.go

```diff
@@ -34,6 +34,23 @@ type Manifest struct {
 type PolicyEntry struct {
 	SANs             []string
 	WorkloadSecretID string `json:",omitempty"`
+	Role             Role   `json:",omitempty"`
+}
+
+// Validate checks the validity of a policy entry given its policy hash.
+func (e PolicyEntry) Validate(policyHash HexString) error {
+	if _, err := policyHash.Bytes(); err != nil {
+		return fmt.Errorf("decoding policy hash %s: %w", policyHash, err)
```

Comment:
```suggestion
		return fmt.Errorf("decoding policy hash %q: %w", policyHash, err)
```
Nit: that should make whitespace errors or similar clearer.